config/src/config/consensus_config.rs

```diff
@@ -13,9 +13,12 @@ use serde::{Deserialize, Serialize};
 use std::path::PathBuf;
 
 // NOTE: when changing, make sure to update QuorumStoreBackPressureConfig::backlog_txn_limit_count as well.
-const MAX_SENDING_BLOCK_TXNS: u64 = 1900;
+const MAX_SENDING_BLOCK_UNIQUE_TXNS: u64 = 1900;
 pub(crate) static MAX_RECEIVING_BLOCK_TXNS: Lazy<u64> =
-    Lazy::new(|| 10000.max(2 * MAX_SENDING_BLOCK_TXNS));
+    Lazy::new(|| 10000.max(2 * MAX_SENDING_BLOCK_UNIQUE_TXNS));
+static MAX_SENDING_BLOCK_TXNS: Lazy<u64> =
+    Lazy::new(|| MAX_SENDING_BLOCK_UNIQUE_TXNS.max(MAX_RECEIVING_BLOCK_TXNS.saturating_sub(2000)));
```

Comment:
`2000` is a magic number. can we at least add a comment about what this cushion is?

Reply:
Added comments.

Reply:
I think MAX_RECEIVING_BLOCK_TXNS is just intentionally huge, and we can bump it arbitrarily, I wouldn't make other constants depend on it.

I suggest putting here something more realistic - like 4-5k, unless we think we actually need 8k


we can just assert it is lower than MAX_RECEIVING_BLOCK_TXNS